js/binance.js

```diff
@@ -830,7 +830,11 @@ module.exports = class binance extends Exchange {
                 let success = this.safeValue (response, 'success', true);
                 if (!success) {
                     if ('msg' in response)
-                        response = JSON.parse (response['msg']);
+                        try {
+                            response = JSON.parse (response['msg']);
+                        } catch (e) {
+                            response = '';
```

Comment:
Should this line be like response = {} ? ;)

Reply:
yes

Reply:
In fact, I think of moving it to the base class, because if `.parseJsonResponse` is `true` (if we expect JSON from a particular endpoint) and the exchange replies with a non-json response, then we should throw an ExchangeError, no?

Reply:
Nope, ignore the previous comment on ExchangeError, the easiest way to handle this would be like you did. Sorry for the confusion, just remembered that handleErrors gets called first...

Reply:
Yeah, pretty sure it is already in the base class (at least in python) - https://github.com/ccxt/ccxt/blob/239e591890600ad8058743c0153663e6fb6b698e/python/ccxt/base/exchange.py#L409

Also this is a very specific situation because it is nested json (a json string inside an already json response)